bufferevent_ssl.c

```diff
@@ -268,31 +269,38 @@ do_read(struct bufferevent_ssl *bev_ssl, int n_to_read) {
 	if (n < 0)
 		return OP_ERR;
 
-	for (i=0; i<n; ++i) {
+	for (i = 0; i < n;) {
 		if (bev_ssl->bev.read_suspended)
 			break;
 		bev_ssl->ssl_ops->clear_error();
-		r = bev_ssl->ssl_ops->read(bev_ssl->ssl, space[i].iov_base, space[i].iov_len);
-		if (r>0) {
+		r = bev_ssl->ssl_ops->read(
+			bev_ssl->ssl, space[i].iov_base + len, space[i].iov_len - len);
```

Comment:
You've lost my changes

```suggestion
			bev_ssl->ssl, (unsigned char *)space[i].iov_base + len, space[i].iov_len - len);
```